service/utils.go

```diff
@@ -88,24 +88,24 @@ func catalogResource(catalog model.Catalog, apiContext *api.ApiContext, envId st
 	}
 }
 
-func templateDefaultVersion(template model.Template, catalogName string, apiContext *api.ApiContext) (string, string) {
+func templateDefaultVersion(template model.Template, catalogName string, apiContext *api.ApiContext) (string, string, string) {
 	var defaultTemplateVersionId string
 	for _, version := range template.Versions {
 		if version.Version == template.DefaultVersion {
 			defaultTemplateVersionId = generateVersionId(catalogName, template, version)
 			selfLink := apiContext.UrlBuilder.ReferenceByIdLink("templates", defaultTemplateVersionId)
-			return selfLink, defaultTemplateVersionId
+			return selfLink, defaultTemplateVersionId, version.Version
 		}
 	}
 
 	sort.Sort(model.Versions(template.Versions))
 	if len(template.Versions) != 0 {
 		defaultTemplateVersionId = generateVersionId(catalogName, template, template.Versions[len(template.Versions)-1])
 		selfLink := apiContext.UrlBuilder.ReferenceByIdLink("templates", defaultTemplateVersionId)
-		return selfLink, defaultTemplateVersionId
+		return selfLink, defaultTemplateVersionId, template.Versions[len(template.Versions)-1].Version
```

Comment:
And this is where it is detected automatically? This takes min/max Rancher version into account, right?

Reply:
Tested it out, it does take min/max Rancher version into account.